stable/yugabyte/templates/service.yaml

```diff
@@ -205,10 +187,21 @@ spec:
             podAffinityTerm:
               labelSelector:
                 matchExpressions:
+                {{- if $root.Values.oldNamingStyle }}
                 - key: app
                   operator: In
                   values:
                   - "{{ .label }}"
+                {{- else }}
+                - key: app.kubernetes.io/name
+                  operator: In
+                  values:
+                  - "{{ .label }}"
+                - key: release
+                  operator: In
+                  values:
+                  - {{ $root.Release.Name | quote }}
```

Comment:
Having the release label here means that two YugabyteDB pods can be on same node if they are part of different releases. _Also some pod with old naming scheme can be with a pod of new naming scheme._